hazelcast-wm/src/main/java/com/hazelcast/web/ClusteredSessionService.java

```diff
@@ -0,0 +1,605 @@
+            Object value = sss.getSerializationService().toObject(entry.getValue());
+            attributes.add(new MapEntrySimple<String, Object>(key, value));
+        }
+        return attributes;
+    }
+
+    Object getAttribute(String sessionId, String attributeName) throws Exception {
+        GetAttribute entryProcessor = new GetAttribute(attributeName);
+        entryProcessor.setJvmId(jvmId);
+        return executeOnKey(sessionId, entryProcessor);
+    }
+
+    void deleteAttribute(String sessionId, String attributeName) throws Exception {
+        setAttribute(sessionId, attributeName, null);
+    }
+
+    void setAttribute(String sessionId, String attributeName, Object value) throws Exception {
+        Data dataValue = (value == null) ? null : sss.getSerializationService().toData(value);
+        SessionUpdateProcessor sessionUpdateProcessor = new SessionUpdateProcessor(attributeName, dataValue);
+        sessionUpdateProcessor.setJvmId(jvmId);
+        executeOnKey(sessionId, sessionUpdateProcessor);
+    }
+
+    /**
+     * @param sessionId  sessionId
+     * @param invalidate if true remove the distributed session, otherwise just
+     *                   remove the jvm reference
+     */
+    public boolean deleteSession(String sessionId, boolean invalidate) {
+        try {
+            doDeleteSession(sessionId, invalidate);
+            return true;
+        } catch (Exception e) {
+            orphanSessions.add(sessionId);
+            return false;
+        }
+    }
+
+    private void doDeleteSession(String sessionId, boolean invalidate) throws Exception {
+        DeleteSession entryProcessor = new DeleteSession(sessionId, invalidate);
+        entryProcessor.setJvmId(jvmId);
+        executeOnKey(sessionId, entryProcessor);
+    }
+
+    public Set<String> getAttributeNames(String id) throws Exception {
+        return (Set<String>) executeOnKey(id, new GetAttributeNames());
+    }
+
+    public void updateAttributes(String id, Map<String, Object> updates) throws Exception {
+        SerializationService ss = sss.getSerializationService();
+        SessionUpdateProcessor sessionUpdate = new SessionUpdateProcessor(updates.size());
+        sessionUpdate.setJvmId(jvmId);
+        for (Map.Entry<String, Object> entry : updates.entrySet()) {
+            String name = entry.getKey();
+            Object value = entry.getValue();
+            sessionUpdate.attributes.put(name, ss.toData(value));
+        }
+        executeOnKey(id, sessionUpdate);
+    }
+
+    public void destroy() {
+        if (hazelcastInstance != null) {
+            try {
+                hazelcastInstance.getLifecycleService().shutdown();
+                ensureInstanceRunning = false;
+                es.shutdown();
+            } catch (Exception ignored) {
+                EmptyStatement.ignore(ignored);
+            }
+        }
+    }
+
+    public static class GetAttribute implements EntryProcessor<String, SessionState>,
```

Comment:
Better to remove all these inner classes. I would suggest making a directory called com.hazelcast.web.entryprocessors and move all those EntryProcessor implementations in that package.